src/System.Net.Sockets/tests/FunctionalTests/TcpListenerTest.cs

```diff
@@ -142,6 +142,74 @@ public void IPv6_Only_Works()
             s.Close();
         }
 
+        [Fact]
+        public async Task Accept_StartAfterStop_AcceptsSuccessfully()
+        {
+            var listener = new TcpListener(IPAddress.Loopback, 0);
+            listener.Start();
+            await VerifyAccept(listener);
+            listener.Stop();
+
+            Assert.Null(listener.Server);
```

Comment:
Our [documentation for Stop](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.tcplistener.stop?view=netcore-3.0#notes-to-callers) states:

> The Stop() method also closes the underlying Socket, and creates a new Socket for the TcpListener.

So, `listener.Server` should not be `null` here. To maintain observable behavior, the `Server` property needs to create the socket if `_serverSocket` is `null`.

Reply:
Delaying a new socket creation until the next call to Start method is the essence of this PR whose main purpose is to prevent a socket from leaking on Stop. You are right that it will lead to having Server property being null after stop, however in the issue discussion @karelz [said that it's acceptable](https://github.com/dotnet/corefx/issues/26170#issuecomment-355461859)
> The only observable difference would be that `Server` property would return `null` after `Stop`, which it didn't in the past. We could compensate for that as well, but that looks like overkill to me.

On the other hand, I agree that now it contradicts with the documentation, so I'm thinking if we can update the documentation for the next .Net Core release to match the new behavior. What do you think?

Reply:
I agree with Cory. Why not just create the Socket if the property is accessed? If someone is doing that after they call Stop, they obviously want it for something. And if they don't, then the PR's benefits remain.

Reply:
Ok, I will implement an on-demand init.

Reply:
Fixed